### ISSUE TITLE
Empty array as default parameter doesn't work as expected

### ISSUE THREAD

**Issue by Lanayx** [dotnet/fsharp#10117]
Please provide a succinct description of the issue.

**Repro steps**

```
type C() =
    member _.M ([<Optional; DefaultParameterValue([||]:byte[])>]keyBytes:byte[]) = 
        keyBytes
```

**Expected behavior**

The code either passes empty array as default argument or doesn't compile

**Actual behavior**

The code compiles, but parameter is not optional anymore, I can't call M() without parameters [sharplab](https://sharplab.io/#v2:DYLgZgzgNAJiDUAfA9gBwKYDsAEBlAnhAC7oC2AsAFBpZ6EmkB0ASgK6ZECWp6jAkh3QAnNLmEA3TgGN0EKlSL4M2AMIAKAJTYAvFWz7sPUgCNh2APqMAstjUBtADwB5VF2SYAhsADc2ACLoYB6swEQACh5CHjwkQgBqXqzo9oiIALogxvgkdmkaAHxpANbo+ABC2bKZlbla2th6Bk0l5ZVylE1GpkLYRAAWnBCMAGKaOo1N+v2D1ppAA===)

**Known workarounds**

Use null instead of empty array. Fsharp-style optional argument won't work, since the method is exposed from library

**Related information**

.NET Core SDK (reflecting any global.json):
 Version:   3.1.101
 Commit:    b377529961

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows

**Comment by vzarytovskii** [comment]
Yeah, we don't process arrays (or rather, we process them as default case: [src/fsharp/TypedTreeOps.fs:L8980](https://github.com/dotnet/fsharp/blob/main/src/fsharp/TypedTreeOps.fs#L8980) - and reurn None as default value), so we have None as default value for ILParameter: [src/fsharp/IlxGen.fs:L5945](https://github.com/dotnet/fsharp/blob/main/src/fsharp/IlxGen.fs#L5945), resulting in not generating .param value for IL method.

I don't think, C# allows this either. I guess we should emit warning?

**Comment by Lanayx** [comment]
@vzarytovskii I think it should raise error  [FS0267] This is not a valid constant expression or custom attribute value

**Comment by cartermp** [comment]
Yeah, this should raise an error. In C# an error is raised unless it is a literal or `null` passed in. There are two issues here, really:

* this one, which should raise an error
* https://github.com/dotnet/fsharp/issues/8551
